src/liballoc/range.rs

```diff
@@ -63,9 +63,74 @@ pub trait RangeArgument<T: ?Sized> {
     /// # }
     /// ```
     fn end(&self) -> Bound<&T>;
-}
 
-// FIXME add inclusive ranges to RangeArgument
+    /// Returns `true` if `item` is contained in the range.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(range_contains)]
+    /// #![feature(collections_range)]
+    ///
+    /// use std::collections::range::RangeArgument;
+    ///
+    /// assert!(!(3..5).contains(&2));
+    /// assert!( (3..5).contains(&3));
+    /// assert!( (3..5).contains(&4));
+    /// assert!(!(3..5).contains(&5));
+    ///
+    /// assert!(!(3..3).contains(&3));
+    /// assert!(!(3..2).contains(&3));
+    ///
+    /// assert!(!(3..).contains(&2));
+    /// assert!( (3..).contains(&3));
+    /// assert!( (3..).contains(&1_000_000_000));
+    ///
+    /// assert!( (..5).contains(&-1_000_000_000));
+    /// assert!( (..5).contains(&4));
+    /// assert!(!(..5).contains(&5));
+    ///
+    /// assert!(!(3..=5).contains(&2));
+    /// assert!( (3..=5).contains(&3));
+    /// assert!( (3..=5).contains(&4));
+    /// assert!( (3..=5).contains(&5));
+    /// assert!(!(3..=5).contains(&6));
+    ///
+    /// assert!( (3..=3).contains(&3));
+    /// assert!(!(3..=2).contains(&3));
+    ///
+    /// assert!( (..=5).contains(&-1_000_000_000));
+    /// assert!( (..=5).contains(&5));
+    /// assert!(!(..=5).contains(&6));
+    /// ```
+    #[unstable(feature = "range_contains", reason = "recently added as per RFC", issue = "32311")]
+    fn contains<U>(&self, item: &U) -> bool
+    where
+        U: ?Sized + PartialOrd<T>,
```

Comment:
Something I don't understand coherence well enough to answer: how does `U: PartialOrd<T>` as a bound compare to `T: PartialOrd<U>`?  Would we prefer to make it easier for `T` or `U` to be a custom type, and which one makes that easier?  Should this just bound both of them for now?